starlite/router.py

```diff
@@ -161,16 +161,24 @@ def register(self, value: ControllerRouterHandler) -> List["BaseRoute"]:
         for route_path, handler_or_method_map in self._map_route_handlers(value=validated_value):
             path = join_paths([self.path, route_path])
             if isinstance(handler_or_method_map, WebsocketRouteHandler):
-                route: "BaseRoute" = WebSocketRoute(path=path, route_handler=handler_or_method_map)
+                route: Union["WebSocketRoute", "ASGIRoute", "HTTPRoute"] = WebSocketRoute(
+                    path=path, route_handler=handler_or_method_map
+                )
                 self.routes.append(route)
             elif isinstance(handler_or_method_map, ASGIRouteHandler):
                 route = ASGIRoute(path=path, route_handler=handler_or_method_map)
                 self.routes.append(route)
             else:
-                existing_handlers: List[HTTPRouteHandler] = list(self.route_handler_method_map.get(path, {}).values())  # type: ignore
+                existing_handlers = self.route_handler_method_map.get(path, {})
+
+                if not isinstance(existing_handlers, dict):
+                    raise ImproperlyConfiguredException(
+                        "Cannot have both HTTP routes and websocket / asgi route handlers on the same path"
+                    )
+
                 route_handlers = unique(list(handler_or_method_map.values()))
```

Comment:
Redundant call to `list`
```suggestion
                route_handlers = unique(handler_or_method_map.values())
```